quickwit/quickwit-metastore/src/metastore/index_metadata/mod.rs

```diff
@@ -126,16 +126,16 @@ impl IndexMetadata {
         Ok(mutation_occurred)
     }
 
-    /// Deletes a source from the index. Returns whether the index was modified (true).
-    pub(crate) fn delete_source(&mut self, source_id: &str) -> MetastoreResult<bool> {
+    /// Deletes a source from the index.
+    pub(crate) fn delete_source(&mut self, source_id: &str) -> MetastoreResult<()> {
```

Comment:
I removed the returned mutation state flag as it forces callers to handle cases that never occur.

Reply:
Interesting. Usually, we err on the side of returning `Ok` if the target of the deletion does not exist or no longer exists. This makes those calls idempotent.